models/project-extension/src/generator.ts

```diff
@@ -0,0 +1,39 @@
+import { fake, Generator } from '@commercetools-test-data/core';
+import { CustomApplicationInstallation } from '@commercetools-test-data/custom-application';
+import { CustomViewInstallation } from '@commercetools-test-data/custom-view';
+import { createRelatedDates } from '@commercetools-test-data/utils';
+
+import * as CategoryRecommendationSettings from './category-recommendation-settings';
+import * as ImageRegex from './image-regex';
+import * as SampleDataImport from './sample-data-import';
+
+import type { TProjectExtension } from './types';
+
+const [getOlderDate, getNewerDate] = createRelatedDates();
+
+const generator = Generator<TProjectExtension>({
+  fields: {
+    createdAt: fake(getOlderDate),
+    updatedAt: fake(getNewerDate),
+    id: fake((f) => f.string.uuid()),
+    projectKey: fake((f) => f.lorem.slug()),
+    imageRegex: fake(() => ImageRegex.random()),
+    applications: [], // Note: This is deprecated
+    installedApplications: fake(() => [CustomApplicationInstallation.random()]),
+    installedCustomViews: fake(() => [CustomViewInstallation.random()]),
+    isRichTextEditorEnabled: fake((f) => f.datatype.boolean()),
```

Comment:
Nit: since we're here, we can also add the new field `isCustomerEmailValidationDisabled` to the model (see https://github.com/commercetools/merchant-center-services/pull/3517)

Reply:
Updated [6f6ec89](https://github.com/commercetools/test-data/pull/562/commits/6f6ec895ccd25330514d2a9db32ff2e674923ee5)